PYTHON_SUPPORT_GUIDE.md

```diff
@@ -0,0 +1,63 @@
+# Python Support Guide
+
+This document walks developers on how to add support for new major & minor Python versions in OpenSearch-Benchmark.
+
+### Add New Python Versions to OpenSearch-Benchmark
+
+1. Add new python major version as tuple in list in [setup.py](https://github.com/opensearch-project/opensearch-benchmark/blob/main/setup.py#L47). For example, if you were to want to add support for python 3.12 to this, you would just add it at the end of this list as a tuple.
```

Comment:
Add a new Python major version as a tuple in this list in..

It would be appropriate to capitalize "Python" in text in other places as well.  Also, change Opensearch-benchmark in text to "OpenSearch Benchmark".